integration-tests/src/test/kotlin/io/spine/examples/protodata/hello/ApplySizeOptionPluginTest.kt

```diff
@@ -0,0 +1,213 @@
+            createValidContactBuilder()
+                .validateAddressCount()
+                .validateEmailCount()
+                .validatePhoneCount()
+
+            val invalidContactBuilder = createInvalidContactBuilder()
+            assertThrows<IllegalStateException> {
+                invalidContactBuilder
+                    .validatePhoneCount()
+            }
+            assertThrows<IllegalStateException> {
+                invalidContactBuilder
+                    .validateAddressCount()
+            }
+            assertThrows<IllegalStateException> {
+                invalidContactBuilder
+                    .validateEmailCount()
+            }
+        }
+
+        @Test
+        fun `by calling field validation methods inside 'build()' method`() {
+            createValidContactBuilder()
+                .build()
+
+            assertThrows<IllegalStateException> {
+                createInvalidContactBuilder()
+                    .build()
+            }
+        }
+    }
+
+    @Test
+    fun `fail the build if 'size' option value is not set`(
+        @TempDir projectDir: File
+    ) {
+
+        val expectedExceptionMessage = "Value of `size` option " +
+                "for field `SizeOptionTest.empty_expression_field` is not set."
+
+        assertBuildFailed(
+            projectDir,
+            File(EMPTY_EXPRESSION_VALUE_PROTO),
+            expectedExceptionMessage
+        )
+    }
+
+    @Test
+    fun `fail the build if 'size' option is applied to non-repeated field`(
+        @TempDir projectDir: File
+    ) {
+
+        val expectedExceptionMessage = "Field " +
+                "`SizeOptionTest.non_repeated_field` is non-repeated " +
+                "and therefore cannot be validated with `size` option."
+
+        assertBuildFailed(
+            projectDir,
+            File(NON_REPEATED_FIELD_PROTO),
+            expectedExceptionMessage
+        )
+    }
+
+    private fun assertBuildFailed(
+        projectDir: File,
+        protoSourceFile: File,
+        expectedExceptionMessage: String
+    ) {
+
+        val project = GradleProject.setupAt(projectDir)
+            .fromResources(TEST_PROJECT_DIR)
+            .copyBuildSrc()
+            .create()
+
+        val stderr = StringWriter()
+        (project.runner as DefaultGradleRunner)
+            .withJvmArguments(
+                "-Xmx4g",
+                "-XX:MaxMetaspaceSize=512m",
+                "-XX:+HeapDumpOnOutOfMemoryError"
+            )
+            .forwardStdError(stderr)
+
+        protoSourceFile.copyTo(File(projectDir, PROTO_MODEL_FILE), true)
+
+        try {
+            project.executeTask(McJavaTaskName.launchProtoData)
+            fail("The build is unexpectedly successful.")
+        } catch (_: UnexpectedBuildFailure) {
+        }
+
+        // Uncomment the lines below to see the content of stderr stream.
+        //File(protoSourceFile.name + "_stderr.log")
```

Comment:
I would drop this piece altogether. This may be a part of the test documentation.